link/src/main/java/com/stripe/android/link/LinkActivity.kt

```diff
@@ -19,30 +25,47 @@ import com.stripe.android.link.ui.paymentmenthod.PaymentMethodScreen
 import com.stripe.android.link.ui.signup.SignUpScreen
 import com.stripe.android.link.ui.verification.VerificationScreen
 import com.stripe.android.link.ui.wallet.WalletScreen
+import com.stripe.android.ui.core.CircularProgressIndicator
 
-internal class LinkActivity : AppCompatActivity() {
-    @VisibleForTesting
-    internal var viewModelFactory: ViewModelProvider.Factory = LinkActivityViewModel.Factory()
-    private val viewModel: LinkActivityViewModel by viewModels { viewModelFactory }
+internal class LinkActivity : ComponentActivity() {
+    internal var viewModel: LinkActivityViewModel? = null
 
     @VisibleForTesting
     internal lateinit var navController: NavHostController
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
+        try {
+            viewModel = ViewModelProvider(this, LinkActivityViewModel.factory())[LinkActivityViewModel::class.java]
+        } catch (e: NoArgsException) {
+            Log.e(TAG, "Failed to create LinkActivityViewModel", e)
```

Comment:
can you use our [Logger](https://github.com/stripe/stripe-android/blob/fff2e2534332325431990163c723a72e0007b83c/stripe-core/src/main/java/com/stripe/android/core/Logger.kt#L7) class here instead of Log? Logger helps us avoid creating noisy logs in user apps.